pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -261,9 +289,26 @@ func getServerByName(client *gophercloud.ServiceClient, name string) (*servers.S
 	if len(serverList) == 0 {
 		return nil, ErrNotFound
 	} else if len(serverList) > 1 {
-		return nil, ErrMultipleResults
+		var lastMatch *servers.Server
+		// try to find exactMatch
+		for _, server := range serverList {
+			if server.Name == exactMatch {
```

Comment:
That's quite different to GCE and maybe others. Kubernetes assume that instance name is unique and all the methods operate with names as identifiers of nodes. We could use uuid as instance name, but that would require basically to refactor whole OpenStack cloud provider.